core/src/switcher/switcher.tsx

```diff
@@ -27,17 +84,23 @@ export const Switcher = <T,>(props: Props<T>): JSX.Element => (
 				<Button
 					key={option.label || option.key}
 					icon={option.icon}
+					iconLabel={option.iconLabel}
 					children={option.label}
 					onClick={() => {
 						if (selected === false) props.setValue(option.value);
 					}}
-					selected={selected}
-					disabled={option.disabled}
+					disabled={props.disabled || option.disabled}
 					fill={props.fill}
 					size={props.size}
 					style={props.style}
+					{...(props.highlight
+						? { highlight: selected }
+						: { selected: selected })}
 				/>
 			);
 		})}
 	</ButtonGroup>
 );
+
+Switcher.styles = Button.styles;
+Switcher.sizes = Button.sizes;
```

Comment:
https://github.com/moaijs/moai/pull/151#pullrequestreview-642337289

@lqt93 yeah that looks bad :( but I'm not really sure how to fix it. It's a known behaviour from the Button, actually.

As you see, it depends on the content. At first thought, we could check whether it's an icon-only button and use smallIcon instead of small, for example. However, thing is in icon-only cases, both sizes (smallIcon and small) are expected (one makes the button an absolute square, while the other doesn't).

Or maybe, we could throw a runtime error, like "SmallIcon can only be used for icon-only buttons" :))

Anyway, I think it's ok to address in another PR